apps/web/pages/api/cron/changeTimeZone.ts

```diff
@@ -0,0 +1,170 @@
+import type { NextApiRequest, NextApiResponse } from "next";
+
+import dayjs from "@calcom/dayjs";
+import prisma from "@calcom/prisma";
+import { getDefaultScheduleId } from "@calcom/trpc/server/routers/viewer/availability/util";
+
+const travelScheduleSelect = {
+  id: true,
+  startDate: true,
+  endDate: true,
+  timeZone: true,
+  prevTimeZone: true,
+  user: {
+    select: {
+      id: true,
+      timeZone: true,
+      defaultScheduleId: true,
+    },
+  },
+};
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+  const apiKey = req.headers.authorization || req.query.apiKey;
+  if (process.env.CRON_API_KEY !== apiKey) {
+    res.status(401).json({ message: "Not authenticated" });
+    return;
+  }
+
+  if (req.method !== "POST") {
+    res.status(405).json({ message: "Invalid method" });
+    return;
+  }
+
+  let timeZonesChanged = 0;
+
+  const setNewTimeZone = async (timeZone: string, user: { id: number; defaultScheduleId: number | null }) => {
+    await prisma.user.update({
+      where: {
+        id: user.id,
+      },
+      data: {
+        timeZone: timeZone,
+      },
+    });
+
+    const defaultScheduleId = await getDefaultScheduleId(user.id, prisma);
+
+    if (!user.defaultScheduleId) {
+      // set default schedule if not already set
+      await prisma.user.update({
+        where: {
+          id: user.id,
+        },
+        data: {
+          defaultScheduleId,
+        },
+      });
+    }
+
+    await prisma.schedule.updateMany({
+      where: {
+        id: defaultScheduleId,
+      },
+      data: {
+        timeZone: timeZone,
+      },
+    });
```

Comment:
I don't see any messaging about updating the default schedule timezone. I have two types of availabilities that I use for different event types. What if a user wants all their schedules updated? Could be a @ciaranha @itsberkaya question.

Reply:
@ciaranha and me talked about this a while ago. When changing the profile timezone only the default schedule should be updated. I think it actually would make sense to update the default schedule + every schedule that has the same timezone as the old profile timezone